src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt

```diff
@@ -68,11 +68,10 @@ class VisitService(
     val offenderBooking = offenderBookingRepository.findByOffenderNomsIdAndActive(offenderNo, true)
       .orElseThrow(NotFoundException(offenderNo))
 
-    val existingVisits = visitRepository.findByVsipVisitId(VSIP_PREFIX + visitDto.vsipVisitId)
-    if (!existingVisits.isEmpty()) {
-      val existingVisit = existingVisits[0]
+    val existingVisit = visitRepository.findOneByVsipVisitId(VSIP_PREFIX + visitDto.vsipVisitId)
+    if (existingVisit.isPresent) {
```

Comment:
it might be nice to have a repository function returning a kotlin nullable to avoid having to work with optional

Reply:
good point, will do

Reply:
actaully that bit is getting removed along with the VSIP id!